tools/debug/multi_index_helpers.cpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * Put into scipp::core namespace:
```

Comment:
Maybe you can just provide these snippets in individual header files that can be included directly at the relevant places?

Reply:
The problem is that the `MultiIndex` operator needs to be a friend because there are no accessors for most of the internal state.

I considered putting it into the class and surrounding it with `#ifndef NDEBUG` but I would rather not have conditionally compiled pieces of code.

Reply:
My point is, if you want to use this:

```cpp
class MultiIndex:
  // ...
private:
  #include multi_index_print
};
```

Reply:
Then you don't need to copy the code, you just include it.

Reply:
Dirty, but possible. This would also need `operator<<` for array. I can make a general debug IO header which provides that and maybe the same for `std::vector`. Would this still fit into `tools`?

Reply:
I guess, yes.